Watchman.Discord/Areas/Users/Controllers/UsersController.cs

```diff
@@ -44,10 +48,50 @@ public void GetAvatar(DiscordRequest request, Contexts contexts)
         [DiscordCommand("add role")] //todo
         public void AddRole(DiscordRequest request, Contexts contexts)
         {
-            var commandRole = request.OriginalMessage.ToLowerInvariant().Replace("-add role ", string.Empty); //TODO use DiscordRequest properties
-            var safeRoles = this._queryBus.Execute(new GetDiscordServerSafeRolesQuery(contexts.Server.Id)).SafeRoles;
-            var messagesService = _messagesServiceFactory.Create(contexts);
-            _rolesService.AddRoleToUser(safeRoles, messagesService, contexts, commandRole);
+            var args = request.Arguments.Skip(1).ToArray();
+            var rolesToAssign = args.Select(x => x.Value).ToList();
+
+            if (rolesToAssign.Count() < 1)
+            {
+                throw new NotEnoughArgumentsException();
+            }
+
+            if (args.GroupBy(x => x.Value)
+                .Select(x => x.First()).Count() != args.Length)
+            {
+                throw new ArgsAreDuplicatedException();
+            }
+
+            var allRoleNames = _usersRolesService.GetRoles(contexts.Server).ToList().Select(x => x.Name);
+            var safeRoleNames = _queryBus.Execute(new GetDiscordServerSafeRolesQuery(contexts.Server.Id)).SafeRoles.ToList().Select(x => x.Name);
+            var userRoles = contexts.User.Roles.Select(x => x.Name);
+
+            var messageService = _messagesServiceFactory.Create(contexts);
+            bool canBeAssigned = true;
+
+            rolesToAssign.ForEach(roleToCheck =>
+            {
+                if (!allRoleNames.Contains(roleToCheck)
+                || !safeRoleNames.Contains(roleToCheck))
+                {
+                    messageService.SendResponse(x => x.RoleNotFoundOrIsNotSafe(contexts, roleToCheck), contexts);
+                    canBeAssigned = false;
+                }
+                if (userRoles.Contains(roleToCheck))
+                {
+                    messageService.SendResponse(x => x.RoleIsInUserAlready(contexts, roleToCheck), contexts);
+                    canBeAssigned = false;
+                }
+            });
+
+            if (canBeAssigned)
```

Comment:
czemu to nie jest w poprzedniej pętli?
przez to masz dwie osobne pętle 

nie chcemy sytuacji gdzie użytkownik dostaje mase błędów i podstawowa czynność udaje mu sie po 10 minutach prób

myśle że tutaj lepiej byłoby zwrócić komunikat z listą roli które udało się dodać i tych które nie udało się dodać 
jeśli wszystkie udało się dodać, zamiast listy powinna być informacja że udało się dodać wszystkie 
ale => tylko jeśli było ich więcej niż jedna, bo to byłoby bez sensu gdyby ktoś dodawał sobie jedną, a dostawałby informacje jakby dodał ich wiele